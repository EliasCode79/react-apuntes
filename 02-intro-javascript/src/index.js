//  Operador ternario

let msj = 'Este es un msj';

let esVerdad = true

let msjSend = ( esVerdad && es)