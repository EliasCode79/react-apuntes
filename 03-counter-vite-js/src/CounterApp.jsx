import PropTypes from 'prop-types';

export const CounterApp = ({ value }) => {
	const handler = (event) => {
		// console.log(event)
		console.log('+1');
		value = 100;
	};

	return (
		<>
			<h1>Counter App</h1>
			<h2>{value}</h2>

			{/* onclick envia por defecto un event, en react podemos llamar a una funcion si (), cuando se envia un parametro */}
			<button onClick={handler}>+1</button>
		</>
	);
};

CounterApp.propTypes = {
	value: PropTypes.number.isRequired,
};

CounterApp.defaultPros = {
	value: 'No hay numero',
};
