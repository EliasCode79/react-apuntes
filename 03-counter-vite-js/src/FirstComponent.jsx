export const FirtsApp = () => {
	return <h2>Elias Franco</h2>;
};
