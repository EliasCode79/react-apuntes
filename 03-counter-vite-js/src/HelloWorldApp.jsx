// no estamos importando React, porque ya los estamos importando de manera global con Vite
export const App = () => {
	return <h1>Hola mundo</h1>;
};
