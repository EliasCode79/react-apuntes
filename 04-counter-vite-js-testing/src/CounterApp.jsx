import { useState } from 'react';
import PropTypes from 'prop-types';

export const CounterApp = ({ value }) => {
	const [counter, setCounter] = useState(value);

	const handler = (operation) => {
		setCounter(counter + 1);
	};

	const handlerRest = () => setCounter(counter - 1);
	const handlerReset = () => setCounter(value);

	return (
		<>
			<h1>Counter App</h1>
			<h2>{counter}</h2>

			<button onClick={handler}>+1</button>
			<button onClick={handlerRest}>-1</button>
			<button aria-label='btn-reset' onClick={handlerReset}>
				Reset
			</button>
		</>
	);
};

CounterApp.propTypes = {
	value: PropTypes.number.isRequired,
};

CounterApp.defaultPros = {
	value: 'No hay numero',
};
