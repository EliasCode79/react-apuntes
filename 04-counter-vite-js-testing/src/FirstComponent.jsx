import PropTypes from 'prop-types';

export const FirtsApp = ({ title, subTitle, name }) => {
	return (
		<>
			<h1 data-testid='test-title'>{title}</h1>
			<p>{subTitle}</p>
			<p>{name}</p>
		</>
	);
};

// indicamos q tipos de datos esperamos y si es obligatorio o no
FirtsApp.propTypes = {
	title: PropTypes.string.isRequired, // indicamos q es obligatorio q el required.
	subTitle: PropTypes.string,
};

FirtsApp.defaultProps = {
	name: 'no hay nombre',
	// title: 'no hay titulo',
	subTitle: 'no hay subtitulo',
};
