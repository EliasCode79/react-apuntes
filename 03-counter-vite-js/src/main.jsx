import React from 'react';
import ReactDOM from 'react-dom/client';
import { FirtsApp } from './FirstComponent';
import { CounterApp } from './CounterApp';
// import { App } from './HelloWorldApp';
import './styles.css';

ReactDOM.createRoot(document.querySelector('#root')).render(
	<React.StrictMode>
		<CounterApp value={20} />
	</React.StrictMode>
);
