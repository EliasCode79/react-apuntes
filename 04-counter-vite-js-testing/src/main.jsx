import React from 'react';
import ReactDOM from 'react-dom/client';
import { FirtsApp } from './FirstComponent';
import { CounterApp } from './CounterApp';
// import { App } from './HelloWorldApp';
import './styles.css';

ReactDOM.createRoot(document.querySelector('#root')).render(
	<React.StrictMode>
		<FirtsApp title='HOla soy elias' />
		{/* <CounterApp value={20} /> */}
	</React.StrictMode>
);
