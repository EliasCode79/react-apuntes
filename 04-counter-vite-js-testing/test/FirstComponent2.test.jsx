import { render, screen } from '@testing-library/react';
import { FirtsApp } from '../src/FirstComponent';

describe('pruebas en FirstComponent', () => {
	const title = 'hola soy un titulo';
	const subTitle = 'hola soy un subtitulo';

	test('Debe hace un match con el snapshot', () => {
		const { container } = render(<FirtsApp title={title} />);
		expect(container).toMatchSnapshot();
	});

	test('Debe mostrar el msj de: hola soy un titulo', () => {
		// el 'screen' el la pantalla q estamos renderizando.
		// screen.debug();
		render(<FirtsApp title={title} />);
		expect(screen.getByText(title)).toBeTruthy();
		// screen.debug();  // este metodo nos muestra el dom(html) del screen.
	});

	test('Debe mostrar el titulo en un h1', () => {
		render(<FirtsApp title={title} />);
		expect(screen.getByRole('heading', { level: 1 }).innerHTML).toContain(title);
	});

	test('Debe mostrar el subtitulo enviado por los props', () => {
		render(<FirtsApp title={title} subTitle={subTitle} />);

		expect(screen.getAllByText(title).length).toBe(1);
	});
});
