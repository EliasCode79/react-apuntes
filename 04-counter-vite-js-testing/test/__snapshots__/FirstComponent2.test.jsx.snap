// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`pruebas en FirstComponent Debe hace un match con el snapshot 1`] = `
<div>
  <h1
    data-testid="test-title"
  >
    hola soy un titulo
  </h1>
  <p>
    no hay subtitulo
  </p>
  <p>
    no hay nombre
  </p>
</div>
`;
